drivers/ethernet/eth_stm32_hal_common.c

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * Copyright (c) 2020 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ * Copyright (c) 2021 Carbon Robotics
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME eth_stm32_hal
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/crc.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/phy.h>
+#include <ethernet/eth_stats.h>
+#include <soc.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/net/lldp.h>
+#include <zephyr/drivers/hwinfo.h>
+
+#if defined(CONFIG_NET_DSA_DEPRECATED)
+#include <zephyr/net/dsa.h>
+#endif
+
+#include "eth.h"
+#include "eth_stm32_hal_priv.h"
+
+#if DT_INST_PROP(0, zephyr_random_mac_address)
+#define ETH_STM32_RANDOM_MAC
+#endif
+
+#if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
+	    !DT_NODE_HAS_STATUS_OKAY(DT_CHOSEN(zephyr_dtcm))
+#error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
+#endif
+
+#define ST_OUI_B0 0x00
+#define ST_OUI_B1 0x80
+#define ST_OUI_B2 0xE1
+
+#define ETH_STM32_HAL_MTU NET_ETH_MTU
+#define ETH_STM32_HAL_FRAME_SIZE_MAX (ETH_STM32_HAL_MTU + 18)
+
+uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_STM32_RX_BUF_SIZE] __eth_stm32_buf;
+uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_STM32_TX_BUF_SIZE] __eth_stm32_buf;
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32n6_ethernet)
+ETH_DMADescTypeDef dma_rx_desc_tab[ETH_DMA_RX_CH_CNT][ETH_RXBUFNB] ALIGN_32BYTES(__eth_stm32_desc);
+ETH_DMADescTypeDef dma_tx_desc_tab[ETH_DMA_TX_CH_CNT][ETH_TXBUFNB] ALIGN_32BYTES(__eth_stm32_desc);
+#else
+ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __eth_stm32_desc;
+ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __eth_stm32_desc;
+#endif
+
+const struct device *eth_stm32_phy_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, phy_handle));
+
+struct net_if *eth_stm32_get_iface(struct eth_stm32_hal_dev_data *ctx)
+{
+	return ctx->iface;
+}
```

Comment:
having this as a non static function, seems to add overhead, why not remove it and access it directly

Reply:
Done